### PR TITLE
Privileges datasource

### PR DESCRIPTION
**Pull request by appilon**
Datasource that exports list of privileges. See examples for how to use terraform `for` as well as `dynamic` to filter the list, and generate `privileges` blocks in role resources.

This PR also enables `ORG_UNIT` scoped `role_assignment`, however the practitioner must ensure only `is_org_unit_scopable` privileges are passed along.

Odd discovery, the Google API returns `org_unit` ids with a `id:` prefix, somehow this isn't an issue for the org unit resource/datasource itself, however I had to strip that prefix when passing it the `role_assignment` resource. Perhaps the `org_unit` resource should strip that itself? It feels like a mistake on the API end to have that prefix.

Closes #82

### CODE REVIEW

File: docs/resources/group_settings.md [hunk 1]
```diff
@@ -71,6 +72,14 @@ resource "googleworkspace_group_settings" "sales-settings" {
 - **id** (String) The ID of this resource.
 - **name** (String) Name of the group, which has a maximum size of 75 characters.
 
+<a id="nestedblock--timeouts"></a>
+### Nested Schema for `timeouts`
```
Comment (megan07) [inline docs/resources/group_settings.md:76]: oh thanks :)

File: internal/provider/data_source_privileges.go [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+				"service_id":           priv.ServiceId,
+				"etag":                 priv.Etag,
+				"is_org_unit_scopable": priv.IsOuScopable,
+				"privilege_name":       priv.PrivilegeName,
+				"service_name":         priv.ServiceName,
+			})
+			duplicates[id] = true
+		}
+		if len(priv.ChildPrivileges) > 0 {
+			result = append(result, flattenAndPrunePrivileges(priv.ChildPrivileges, duplicates)...)
```
Comment (megan07) [inline internal/provider/data_source_privileges.go:107]: Have you found that there would be a parent duplicated in a child? I'm wondering if we need to be concerned about passing `duplicates` around?

File: docs/data-sources/privileges.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+## Schema
+
+### Optional
+
+- **id** (String) The ID of this resource.
+
+### Read-Only
+
+- **etag** (String) ETag of the resource.
+- **items** (List of Object) A list of Privilege resources. (see [below for nested schema](#nestedatt--items))
```
Comment (megan07) [inline docs/data-sources/privileges.md:57]: do we want to mention somewhere that this will be a flattened list?

**Review by megan07 — APPROVED** [review]
looks great! thanks!

File: internal/provider/data_source_privileges.go (hunk 2 above)
Comment (appilon) [reply internal/provider/data_source_privileges.go:107]: I wasn't sure, but this solution should be robust

File: docs/data-sources/privileges.md (hunk 3 above)
Comment (appilon) [reply docs/data-sources/privileges.md:57]: will do

**Pull request merged**